src/Rings/mpoly-graded.jl

```diff
@@ -2209,13 +2209,46 @@ function homogenization(V::Vector{T},  W::Union{ZZMatrix, Matrix{<:IntegerUnion}
   return [_homogenization(V[i], S, pos) for i=1:l]
 end
 
+
+# To homogenize an ideal we look for some additional simple elements
+# (as redundant generators), and then apply the standard algorithm.
+
+# Internal function: used only in gens_for_homog (immediately below)
+function num_terms(f)
+    return length(collect(coefficients(f)));
+end;
+
```

Comment:
just use `length(f)` this avoids the extremely expensive `collect(coefficients`
Many iterators also support `length`, so even `length(coefficients(f))` would be better

Reply:
Thanks.  I've updated my local copy.